src/arrayMethodJoin.js

```diff
@@ -4,9 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function (separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += '';
+      } else {
+        result += `${this[i]}`;
+      }
+
+      if (i < this.length - 1) {
+        result += separator;
+      }
+    }
+
+    return result;
   };
 }
 
-module.exports = applyCustomJoin;
+module.exports = applyCustomJoin;
\ No newline at end of file
```

Comment:
Files should end with a newline to comply with the POSIX standard. This ensures better compatibility across different systems and tools.